src/main/java/scene/Scene.java

```diff
@@ -1,315 +1,338 @@
+    }
+
+    public final void updateGameObject(GameObject gameObject, boolean insertion) {
+        StaticCollider staticCollider = gameObject.getComponent(StaticCollider.class);
+        if (staticCollider != null && !staticColliders.contains(staticCollider)) {
+            if (insertion)
+                staticColliders.add(staticCollider);
+            else staticColliders.remove(staticCollider);
+        } else {
+            RigidBody rigidBody = gameObject.getComponent(RigidBody.class);
+            if (rigidBody != null && !bodyColliders.contains(rigidBody)) {
+                if (insertion)
+                    bodyColliders.add(rigidBody);
+                else bodyColliders.remove(rigidBody);
+            }
+        }
+    }
+
+    /**
+     * The sceneId is meant to represent the instance of a scene as an integer
+     *
+     * @see SceneManager
+     */
+    public final int sceneId() {
+        return sceneId;
+    }
+
+    /**
+     * @return the List of gameObjects contained in the scene.
+     */
+    public List<GameObject> getGameObjects() {
+        return gameObjects;
+    }
+
+    /**
+     * @param gameObject GameObject to be added.
+     *                   Add a new gameObject to the scene and immediately call its start method.
+     */
+    public void addGameObjectToScene(GameObject gameObject) {
+        gameObjects.add(gameObject);
+        if (active) {
+            gameObject.start();
+            addToRenderers(gameObject);
+        }
+    }
+
+    /**
+     * @param gameObject GameObject to be added.
+     */
+    public void removeGameObjectFromScene(GameObject gameObject) {
+        gameObjects.remove(gameObject);
+        removeFromRenderers(gameObject);
+    }
+
+    /**
+     * Register a renderer to this scene
+     *
+     * @param renderer the renderer to be registered
+     */
+    public void registerRenderer(Renderer<?> renderer) {
+        rendererRegistry.add(renderer);
+    }
+
+    /**
+     * @return Returns the scene's instance of Camera
+     */
+    public Camera camera() {
+        return this.camera;
+    }
+
+    /**
+     * Loops through all the gameObjects in the scene and calls their update methods.
+     */
+    public void updateGameObjects() {
+        for (GameObject go : gameObjects) {
+            go.update(Engine.deltaTime());
+        }
+    }
+
+    public void render() {
+        rendererRegistry.forEach(Renderer::render);
+        lightmapRenderer.render();
+        lightmapRenderer.bindLightmap();
+        renderer.render();
+    }
+
+    public void debugRender() {
+        if (debugMode) this.debugRenderer.render();
+    }
+
+    /**
+     * Initialize all renderers
+     */
+    public void initRenderers() {
+        debugRenderer.init();
+        lightmapRenderer.init();
+        renderer.init();
+        forwardToScreen = new ForwardToTexture(PostProcessStep.Target.DEFAULT_FRAMEBUFFER);
+        forwardToScreen.init();
+    }
+
+    public void addUiObject (Text t) {

```

Comment:
We might change this method, if UI is designed.

Reply:
yes, agreed.